internal/engine/live_update_build_and_deployer.go

```diff
@@ -159,20 +160,40 @@ func (lubad *LiveUpdateBuildAndDeployer) buildAndDeploy(ctx context.Context, cu
 	}()
 
 	if len(toArchive) > 0 {
-		l.Infof("Will copy %d file(s) to container: %s", len(toArchive), cInfo.ContainerID.ShortStr())
+		l.Infof("Will copy %d file(s) to container(s): %s", len(toArchive), cIDStr)
 		for _, pm := range toArchive {
 			l.Infof("- %s", pm.PrettyStr())
 		}
 	}
 
-	err = cu.UpdateContainer(ctx, cInfo, pr, build.PathMappingsToContainerPaths(toRemove), boiledSteps, hotReload)
-	if err != nil {
-		if build.IsUserBuildFailure(err) {
-			return WrapDontFallBackError(err)
+	var lastUserBuildFailure error
+	for _, cInfo := range state.RunningContainers {
```

Comment:
what does this code currently do if there are two different images running in two different containers? can you add a test for this case?

I'm pretty sure that the old code worked correctly, since it would just fall back to the image build codepath if there was more than one container

Reply:
you mean if the user is trying to update their app img and their sidecar img at the same time, or it tilt things that c1 and c2 are both running `appimage:tilt-123` but in fact, one is running that image and another is running `foobar:tilt-456`? or something else?

Reply:
tilt should behave correctly in both cases. it might make sense to add integration tests for them first

Reply:
@nicks bump ☝️

Reply:
So the image target is responsible for any image with name=`appimage` and won't differentiate between in-place updating `appimage:123` and `appimage:456` (see: [we match running containers by name, not name+tag](https://github.com/windmilleng/tilt/blob/3403400979e5db559280dd7782088e040ee69c00/internal/store/build_result.go#L261)). This seems like reasonable behavior to me, but do you think it should be different? (regardless, I'd probs write that test  for `RunningContainersForTarget` rather than for this func)